Automattic/pocket-casts-android
# Replace old sharing code with new one

## Description

Follow-up to #2498. This PR replaces all old sharing code with new code. It also introduces `SharingDialogFragment` to display sharing options.

There are some minor improvements to sharing responsiveness:
* Sharing episode current positions uses live updates from the database.
* The sharing episode file option now shows and hides live based on the database state.
* Switching episodes during playback now updates the episode being shared from the player sheet or closes the sheet if it is a user episode.

I considered testing `ShareViewModel` but decided against it because the logic is simple, as it just combines two streams.

## Testing Instructions

Perform regression tests of sharing. Share from different places in the app:
* Podcast details page
* Episode details page
* Player
* Multi-select
* Swipe left to share
* Bookmarks
* Share downloaded episode

Check if the sharing sheet looks correct with different themes.

## Checklist
 
#### I have tested any UI changes...
